PR: https://github.com/odoo/odoo/pull/

From: 898224f110932addb0e976a3ca60542cda80de25
From: Deep Patel
Date: 2017-09-01 11:02:54

Breaking data model changes scores: del:7 + add:0, change matches:
-    group_product_variant = fields.Boolean("Attributes & Variants",
-    group_pricelist_item = fields.Boolean("Show pricelists to customers",
-    group_product_pricelist = fields.Boolean("Show pricelists On Products",
-    group_uom = fields.Boolean("Units of Measure",
-    group_stock_packaging = fields.Boolean("Packaging", implied_group='product.group_stock_packaging',
-    group_multi_currency = fields.Boolean("Multi-Currencies", implied_group='base.group_multi_currency')
-    module_print_docsaway = fields.Boolean("Docsaway")

Total Changes: 40

[IMP] web,account,...: Regroup settings, add a nav and search bar

Purpose
=======

Settings are often way too long and hard to scan and sometimes you don't know where to find the settings you're looking for.
By adding a left navigation, you can already have an overview of the settings, and switch easily between them.

Specification
=============

* Add a search bar on all the apps that have settings which can search results from all the installed apps.
  (If i'm on sales settings,and I search anything then it shows results from sales settings and also shows
  all the other matching results from all the other apps such as, Inventory...etc) with app name.
* Able to activate feature from the current page results (no matter if the searched result is from another apps).
* Highlight searched word in results
* Delete the sheet, have a full white background
* Add left navigation bar on setting
* Left navigation bar fixed
* List displayed based on installed apps
* On right panel, by default display current app setting and change accordingly
* [Mobile] Left navigation bar displayed on top
* Add Breadcrumb on top of the page: have the name "Settings" + Save / discard CTA + Search
* Add General Settings on the nav bar
* remove "save this page..." notif in all settings
* Delete all recommanded apps section + all checkbox that install app should disappear:
	payroll: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrrSkMzWVpuZ0ZoaFE/view?usp=drivesdk
	Events: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrrYXlHMHd2NTM2blE/view?usp=drivesdk
	Manufacturing: Delete Repair - Quality Control - Maintenance - Product Lifecycle Management [LAP][ok]
	Timesheets: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrrdFg5XzNydkd2TlU/view?usp=drivesdk
	project: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrraWo3NE04TktwcTQ/view?usp=drivesdk
	inventory: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrrQkRaVUR5ekFQYTQ/view?usp=drivesdk
	recruitment: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrrbDdqLWQweF80UkE/view?usp=drivesdk
	purchase: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrrRlAyUDBnb0trQWs/view?usp=drivesdk
	email marketing: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrrOUZONjhvX2k3Y2s/view?usp=drivesdk
	expenses: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrrR0JpN1ZpQmtxU3M/view?usp=drivesdk
	attendances: https://drive.google.com/a/odoo.com/file/d/0B1uIL9E_zXrralpjZ1VJclNpMWs/view?usp=drivesdk
* About duplicate settings:
	Docsaway: Delete from Sales + delete Default Print Provider
	Attributs & Variants: Delete from Purchases - Manufacturing
	Multi-currencies: delete from Sales
	Unit Of Measures: Delete from Purchase & Expenses
* Add a scroll bar on the left bar when there's too many apps
* Keep the navbar visible even when you search
* Move General Settings to the bottom of the list
* Update on Settings:
    - Accounting:
        - rename automatic rates
        - Anglo Saxon Account: Should be in technical feature
        - Place Accounting Reports section before Taxes section
    - CRM:
        - Phone Validation: Enforce international format becomes
        - Local Numbers: (2 radio buttons proposals) Add international prefix / No prefix
    - Fleet: typo " ... a new car if ..." + text is too long <br>
    - project: fix tooltip for colab pads

================================= pseudo patch: =================================

--- a/addons/sale/models/sale_config_settings.py
+++ b/addons/sale/models/sale_config_settings.py
@@ -5,7 +5,6 @@ from odoo import api, fields, models
 
 
 class SaleConfigSettings(models.TransientModel):
-    _name = 'sale.config.settings'
     _inherit = 'res.config.settings'
 
     company_id = fields.Many2one('res.company', string='Company', required=True,
@@ -14,22 +13,11 @@ class SaleConfigSettings(models.TransientModel):
     use_sale_note = fields.Boolean(
         string='Default Terms & Conditions',
         oldname='default_use_sale_note')
-    group_product_variant = fields.Boolean("Attributes & Variants",
-        implied_group='product.group_product_variant')
     group_sale_pricelist = fields.Boolean("Use pricelists to adapt your price per customers",
         implied_group='product.group_sale_pricelist',
         help="""Allows to manage different prices based on rules per category of customers.
                 Example: 10% for retailers, promotion of 5 EUR on this product, etc.""")
-    group_pricelist_item = fields.Boolean("Show pricelists to customers",
-        implied_group='product.group_pricelist_item')
-    group_product_pricelist = fields.Boolean("Show pricelists On Products",
-        implied_group='product.group_product_pricelist')
-    group_uom = fields.Boolean("Units of Measure",
-        implied_group='product.group_uom')
     group_discount_per_so_line = fields.Boolean("Discounts", implied_group='sale.group_discount_per_so_line')
-    group_stock_packaging = fields.Boolean("Packaging", implied_group='product.group_stock_packaging',
-        help="""Ability to select a package type in sales orders and
-                to force a quantity that is a multiple of the number of units per package.""")
     module_sale_margin = fields.Boolean("Margins")
     group_sale_layout = fields.Boolean("Sections on Sales Orders", implied_group='sale.group_sale_layout')
     group_warning_sale = fields.Boolean("Warnings", implied_group='sale.group_warning_sale')
@@ -79,7 +67,6 @@ class SaleConfigSettings(models.TransientModel):
         ('b2c', 'Free sign up (B2C)'),
     ], string='Customer Account')
 
-    group_multi_currency = fields.Boolean("Multi-Currencies", implied_group='base.group_multi_currency')
     module_delivery = fields.Boolean("Shipping Costs")
     module_delivery_dhl = fields.Boolean("DHL")
     module_delivery_fedex = fields.Boolean("FedEx")
@@ -87,7 +74,6 @@ class SaleConfigSettings(models.TransientModel):
     module_delivery_usps = fields.Boolean("USPS")
     module_delivery_bpost = fields.Boolean("bpost")
 
-    module_print_docsaway = fields.Boolean("Docsaway")
     module_product_email_template = fields.Boolean("Specific Email")
     module_sale_coupon = fields.Boolean("Coupons & Promotions")
 
@@ -99,6 +85,19 @@ class SaleConfigSettings(models.TransientModel):
             })
         self.sale_pricelist_setting = self.multi_sales_price and self.multi_sales_price_method or 'fixed'
 
+    @api.onchange('sale_show_tax')
+    def _onchange_sale_tax(self):
+        if self.sale_show_tax == "subtotal":
+            self.update({
+                'group_show_price_total': False,
+                'group_show_price_subtotal': True,
+            })
+        else:
+            self.update({
+                'group_show_price_total': True,
+                'group_show_price_subtotal': False,
+            })
+
     @api.onchange('sale_pricelist_setting')
     def _onchange_sale_pricelist_setting(self):
         if self.sale_pricelist_setting == 'percentage':
@@ -120,19 +119,6 @@ class SaleConfigSettings(models.TransientModel):
                 'group_pricelist_item': False,
             })
 
-    @api.onchange('sale_show_tax')
-    def _onchange_sale_tax(self):
-        if self.sale_show_tax == "subtotal":
-            self.update({
-                'group_show_price_total': False,
-                'group_show_price_subtotal': True,
-            })
-        else:
-            self.update({
-                'group_show_price_total': True,
-                'group_show_price_subtotal': False,
-            })
-
     @api.model
     def get_values(self):
         res = super(SaleConfigSettings, self).get_values()
